File: AtCoder/AtCoder Beginner Contest/049/C/Ruby/ABC_049_C_01.rb
S = gets.chomp

loop do
    size_old = S.length
    ["dreamer", "eraser", "dream", "erase"].each do |i|
        S.chomp!(i)
    end
    size_new = S.length
    break if size_old - size_new == 0
end

puts S.length == 0 ? "YES" : "NO"

File: AtCoder/AtCoder Beginner Contest/049/C/Go/ABC_049_C_01.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var S string
	fmt.Scanf("%s", &S)

	for _, v := range []string{"eraser", "erase", "dreamer", "dream"} {
		S = strings.Replace(s, v, "", -1)
	}

	if len(S) == 0 {
		fmt.Printf("YES")
	} else {
		fmt.Printf("NO")
	}
}